packages/web3/src/contract/dapp-tx-builder.test.ts

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { randomContractAddress, randomContractId, testAddress } from '@alephium/web3-test'
+import { DappTransactionBuilder, genArgs } from './dapp-tx-builder'
+import {
+  AddressConst,
+  BytesConst,
+  ConstFalse,
+  ConstTrue,
+  I256Const,
+  I256Const1,
+  I256ConstN1,
+  U256Const1,
+  U256Const2
+} from '../codec'
+import { base58ToBytes, hexToBinUnsafe } from '../utils'
+import { lockupScriptCodec } from '../codec/lockup-script-codec'
+import { ALPH_TOKEN_ID, ONE_ALPH } from '../constants'
+
+describe('dapp-tx-builder', function () {
+  it('should gen code for args', () => {
+    expect(genArgs(['1i', '2u', '-1', '2'])).toEqual([I256Const1, U256Const2, I256ConstN1, U256Const2])
+
+    expect(genArgs([false, 1n, -1n, '0011', testAddress])).toEqual([
+      ConstFalse,
+      U256Const1,
+      I256ConstN1,
+      BytesConst(hexToBinUnsafe('0011')),
+      AddressConst(lockupScriptCodec.decode(base58ToBytes(testAddress)))
+    ])
+
+    expect(genArgs([false, [1n, 2n], ['0011', '2233']])).toEqual([
+      ConstFalse,
+      U256Const1,
+      U256Const2,
+      BytesConst(hexToBinUnsafe('0011')),
+      BytesConst(hexToBinUnsafe('2233'))
+    ])
+
+    expect(genArgs([true, { array0: [1n, 2n], array1: [-1n, -2n] }])).toEqual([
+      ConstTrue,
+      U256Const1,
+      U256Const2,
+      I256ConstN1,
+      I256Const(-2n)
+    ])
```

Comment:
[Nit] Can we test a case where arg is an object?

Reply:
This test is used to test an object. The second element of the array here is an object.